### PR TITLE
DWARF: `rz-bin -dd` download DWARF from `debuginfod` server

### LINKED ISSUE #4191: `rz-bin -PP` should also download DWARF from `debuginfod` server
Currently, only Rizin downloads, but `rz-bin` can already download PDB files for the binary. The symmetrical feature should be provided for DWARF too. Update the help message and `man` page as well.

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

closes #4191

### CODE REVIEW

File: librz/include/rz_bin.h [hunk 1]
```diff
@@ -163,7 +164,7 @@ typedef enum {
 } RzBinLanguage;
 
 #define RZ_BIN_LANGUAGE_MASK(x)       ((x) & ~RZ_BIN_LANGUAGE_BLOCKS)
-#define RZ_BIN_LANGUAGE_HAS_BLOCKS(x) ((x)&RZ_BIN_LANGUAGE_BLOCKS)
```
Comment (XVilka) [inline librz/include/rz_bin.h]: Undo this change to fix formatting CI job

**Review by XVilka — CHANGES_REQUESTED** [review]

File: binrz/man/rz-bin.1 [hunk 2]
```diff
@@ -45,6 +45,8 @@ List classes in header format
 Create [elf, mach0, pe] with Code and Data hexpairs (see -a)
 .It Fl d
 Show debug/dwarf information
+.It Fl d
```
Comment (XVilka) [inline binrz/man/rz-bin.1]: Shouldn't it be `dd` instead?

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
Another higher-level question. Have you considered reusing the same options `-P` and `-PP` instead? I don't think a file could have *both* PDB and DWARF. So, what to download should be essentially decided by the format.

**Comment by XVilka** [comment]
> Another higher-level question. Have you considered reusing the same options `-P` and `-PP` instead? I don't think a file could have _both_ PDB and DWARF. So, what to download should be essentially decided by the format.

Nevermind, lets keep it as is for now.

**Review by XVilka — APPROVED** [review]

File: librz/main/rz-bin.c [hunk 3]
```diff
@@ -245,6 +246,7 @@ static int rzbin_show_help(int v) {
 		       " RZ_BIN_PDBSERVER: e pdb.server       # use alternative PDB server\n"
 		       " RZ_BIN_SYMSTORE:  e pdb.symstore     # path to downstream symbol store\n"
 		       " RZ_BIN_PREFIX:    e bin.prefix       # prefix symbols/sections/relocs with a specific string\n"
+		       " RZ_BIN_DEBUGINFOD_URLS: e bin.dbginfo.debuginfod_urls # use alternative debuginfod server\n"
 		       " RZ_CONFIG:        # sdb config file\n");
```
Comment (wargio) [inline librz/main/rz-bin.c]: maybe is better to shift the spaces of all the other configs..

File: librz/main/rz-bin.c [hunk 4]
```diff
@@ -1302,6 +1311,13 @@ RZ_API int rz_main_rz_bin(int argc, const char **argv) {
 		pdbopts.symbol_store_path = (char *)rz_config_get(core.config, "pdb.symstore");
 		result = rz_bin_pdb_download(core.bin, state.mode == RZ_OUTPUT_MODE_JSON ? state.d.pj : NULL, ismodejson, &pdbopts);
 	}
+	if (action & RZ_BIN_REQ_DEBUGINFOD) {
+		rz_config_set_b(core.config, "bin.dbginfo.debuginfod", true);
+		if ((tmp = rz_sys_getenv("RZ_BIN_DEBUGINFOD_URLS"))) {
```
Comment (wargio) [inline librz/main/rz-bin.c:1316]: maybe set only if is not empty.

**Pull request merged**